server/routes/api.js

```diff
@@ -1073,129 +1089,359 @@ router.post('/capcodeImport', isLoggedIn, function (req, res, next) {
+                        result: 'created'
+                      })
                     })
-                  })
-                  .catch((err) => {
-                    importresults.push({
-                      address: address,
-                      alias: alias,
-                      result: 'failed' + err
+                    .catch((err) => {
+                      importresults.push({
+                        address: address,
+                        alias: alias,
+                        result: 'failed' + err
+                      })
                     })
-                  })
-              } else {
-                //Create new alias if one didn't get returned.
-                return db('capcodes').insert({
-                  id: null,
-                  address: address,
-                  alias: alias,
-                  agency: agency,
-                  color: color,
-                  icon: icon,
-                  ignore: ignore,
-                  pluginconf: pluginconf
+                }
+              })
+              .catch((err) => {
+                importresults.push({
+                  'address': address,
+                  'alias': alias,
+                  'result': 'failed' + err
                 })
-                  .then((result) => {
-                    importresults.push({
-                      address: address,
-                      alias: alias,
-                      result: 'created'
-                    })
-                  })
-                  .catch((err) => {
-                    importresults.push({
-                      address: address,
-                      alias: alias,
-                      result: 'failed' + err
-                    })
-                  })
-              }
+              });
+          };
+          //Gather all the results, format for the frontend and send it back.
+          let results = { "results": importresults }
+          res.status(200)
+          res.json(results)
+          logger.main.debug('Import:' + JSON.stringify(importresults))
+          nconf.set('database:aliasRefreshRequired', 1);
+          nconf.save();
+        } else {
+          throw 'Error parasing CSV header'
+        }
+      })
+      .catch((err) => {
+        res.status(500).send(err)
+        logger.main.error(err)
+      })
+  });
+
+router.route('/user')
+  .get(isAdmin, function (req, res, next) {
+    db.from('users')
+      .select('*')
+      .then((rows) => {
+        res.json(rows);
+      })
+      .catch((err) => {
+        logger.main.error(err);
+        return next(err);
+      })
+  })
+  .post(isAdmin, function (req, res, next) {
+    var username = req.body.username
+    var email = req.body.email
+    db.table('users')
+      .where('username', '=', username)
+      .orWhere('email', '=', email)
+      .first()
+      .then((row) => {
+        if (row) {
+          //add logging
+          res.status(401).send({ 'status': 'error' });
```

Comment:
Should be 401 bad request, and should probably be descriptive error, return the userid maybe?

Reply:
Well, we should probably redo the error handling for all other routes as well. It's a heck of a mess.